grakn-graql/src/main/java/ai/grakn/graql/internal/gremlin/fragment/OutShortcutFragment.java

```diff
@@ -0,0 +1,93 @@
+/*
+ * Grakn - A Distributed Semantic Database
+ * Copyright (C) 2016  Grakn Labs Limited
+ *
+ * Grakn is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * Grakn is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Grakn. If not, see <http://www.gnu.org/licenses/gpl.txt>.
+ *
+ */
+
+package ai.grakn.graql.internal.gremlin.fragment;
+
+import ai.grakn.concept.TypeLabel;
+import ai.grakn.graql.VarName;
+import org.apache.tinkerpop.gremlin.process.traversal.dsl.graph.GraphTraversal;
+import org.apache.tinkerpop.gremlin.structure.Edge;
+import org.apache.tinkerpop.gremlin.structure.Vertex;
+
+import java.util.Optional;
+
+import static ai.grakn.graql.internal.gremlin.fragment.Fragments.applyTypeLabelToTraversal;
+import static ai.grakn.graql.internal.gremlin.fragment.Fragments.displayOptionalTypeLabel;
+import static ai.grakn.util.Schema.EdgeLabel.SHORTCUT;
+import static ai.grakn.util.Schema.EdgeProperty.RELATION_TYPE_LABEL;
+import static ai.grakn.util.Schema.EdgeProperty.ROLE_TYPE_LABEL;
+
+class OutShortcutFragment extends AbstractFragment {
+
+    private final VarName edge;
+    private final Optional<TypeLabel> roleType;
+    private final Optional<TypeLabel> relationType;
+
+    OutShortcutFragment(
+                VarName relation, VarName edge, VarName rolePlayer, Optional<TypeLabel> roleType,
+                Optional<TypeLabel> relationType) {
+            super(relation, rolePlayer, edge);
+            this.edge = edge;
+            this.roleType = roleType;
+            this.relationType = relationType;
+    }
+
+    @Override
+    public void applyTraversal(GraphTraversal<Vertex, Vertex> traversal) {
```

Comment:
It was similar enough that PMD complained. That was part of the reason I extracted `applyTypeLabelToTraversal`. I don't think extracting any more makes sense. It would have to be a higher-order function accepting `GraphTraversal#inE`/`GraphTraversal#outE`, which I feel is overkill for when something is only duplicated once.